packages/tools/pluggable-widgets-tools/configs/rollup-plugin-collect-native-deps.js

```diff
@@ -0,0 +1,123 @@
+import { basename, dirname, extname, join } from "path";
+import { copy, readJson, writeJson } from "fs-extra";
+import { promises } from "fs";
+import { rm } from "shelljs";
+
+export function collectNativeDependencies({ outputDir, externals, widgetName, shouldCopyNodeModules = true }) {
+    let firstRun = true;
+    const nativeDependencies = [];
+    const nodeModulesPath = join(outputDir, "node_modules");
+    return {
+        name: "collect-native-deps",
+        async resolveId(source) {
+            if (source.startsWith(".")) {
+                return null;
+            }
+
+            try {
+                const packageFilePath = require.resolve(`${source}/package.json`);
```

Comment:
Are we sure we are fine with this approach? `require.resolve` resolves in the context of the current package, i.e. pluggable-widgets-tools, and not necessary the widget itself. If it happens that there are multiple versions of the same dependency installed we might get a wrong one. The same applies to logic inside `getTransitiveDependencies`

Reply:
I do agree that we need to find a better way to find the lib directory. Maybe an npmjs lib??

Reply:
We (@tamelbrun and I) saw that it always try to resolve the dependencies from the current package recursively to the parent. So in this case it looks for the dependencies inside each package and then try to resolve in the parent if it doesnt find.